packages/http/src/util.js

```diff
@@ -102,11 +104,11 @@ export function request(method, path, params) {
       resolvedParams.maxRedirections ??
       (resolvedParams.followAllRedirects === false ? 0 : 5);
 
-    const tls = resolvedParams.tls ?? resolvedParams.agentOptions;
+    const tls = tlsConfig ?? resolvedParams.agentOptions;
```

Comment:
This is a breaking change, because any jobs who used to do `get($.url, { tls: $.tlsOptions })` will now stop working

We have to be super careful about not breaking existing user code

We also need to consider that I should be able to a) specify some properties on my credential (like `ca`), but I should also be able to override these config defaults explicitly by passing a `tls` object into the function